src/Microsoft.OData.Client/Serialization/PrimitiveType.cs

```diff
@@ -340,6 +340,7 @@ private static void InitializeTypes()
             RegisterKnownType(typeof(Byte), XmlConstants.EdmByteTypeName, EdmPrimitiveTypeKind.Byte, new ByteTypeConverter(), true);
             RegisterKnownType(typeof(Byte[]), XmlConstants.EdmBinaryTypeName, EdmPrimitiveTypeKind.Binary, new ByteArrayTypeConverter(), true);
             RegisterKnownType(typeof(DateTimeOffset), XmlConstants.EdmDateTimeOffsetTypeName, EdmPrimitiveTypeKind.DateTimeOffset, new DateTimeOffsetTypeConverter(), true);
+            RegisterKnownType(typeof(DateTime), XmlConstants.EdmDateTimeTypeName, EdmPrimitiveTypeKind.DateTimeOffset, new DateTimeTypeConverter(), true);

```

Comment:
DateTime is not a Edm built-in primitive type. The second parameter could be null.